### PR TITLE
Request Logger JsonData Fix

### LINKED ISSUE #3421: Request Logger Update
Handle jsonData via passThru

improve dev-experience and error reporting

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:

 * Ensures Seldon protocol jsonData, strData and binData are not interpreted but passed as is to elastic
 * Adds some units tests
 * Refactors Makefile
 * Adds envs to Dockerfile for GUNICORN_WORKERS AND GUNICORN_THREADS

**Which issue(s) this PR fixes**:

Fixes #3421

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by SachinVarghese** [comment]
Nice, changes look good to me. And returning the elastic response is really useful for debugging.

File: components/seldon-request-logger/Makefile [hunk 1]
```diff
@@ -7,8 +7,8 @@ CLIENT_ID=sd-api
 OIDC_USERNAME=admin@seldon.io
 OIDC_PASSWORD=xxxxxx
 OIDC_SCOPES=openid profile email groups
-DEPLOY_API_HOST=http://xx.xx.xx.xx/seldon-deploy/api/v1alpha1
-OIDC_PROVIDER=http://xx.xx.xx.xx/auth/realms/deploy-realm
+#DEPLOY_API_HOST=http://xx.xx.xx.xx/seldon-deploy/api/v1alpha1
```
Comment (axsaucedo) [inline components/seldon-request-logger/Makefile:10]: Seems comment left out

**Comment by axsaucedo** [comment]
Looks good added a comment for somethign to consider but we can merge as follow refactor
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3423#issuecomment-885202073 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/seldon-request-logger/Makefile (hunk 1 above)
Comment (cliveseldon) [reply components/seldon-request-logger/Makefile:10]: I commented out so as not to assume deploy exists when running locally but should really improve Makefile for the 2 options of running with and without deploy

**Pull request merged**